app/components/contact-page-components/SendMessage.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import React, { useRef, useState } from "react";
+import emailjs from "@emailjs/browser";
+import { ZodError, z } from "zod";
+
+import SendMessageForm from "./SendMessageForm";
+
+interface SendMessageProps {
+  setShowSuccess: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const SendMessage: React.FC<SendMessageProps> = ({ setShowSuccess }) => {
+  const [errorList, setErrorList] = useState("");
+  const [formData, setFormData] = useState({
+    user_name: "",
+    user_email: "",
+    message: "",
+    contact_info: "",
+  });
+
+  const form = useRef<HTMLFormElement>(null);
+
+  const emailFormSchema = z.object({
+    user_name: z.string().nonempty("Name is required"),
+    user_email: z
+      .string()
+      .email("Invalid email format")
+      .nonempty("Email is required"),
+    message: z
+      .string()
+      .min(100, "Message should be at least 100 characters long"),
+    contact_info: z.string().nonempty("Contact information is required"),
+  });
```

Comment:
Currently, emailFormSchema is defined inside the component, which means **_it gets recreated every time the component re-renders_**. You should move it outside of the component.

Reply:
I've moved it to utils